sdks/android/Makefile

```diff
@@ -192,14 +192,14 @@ package: app/assets/asm/monodroid_mini_test.dll
 
 ## Package Mono.Debugger.Soft test assemblies
 
-.PHONY: $(TOP)/mcs/class/Mono.Debugger.Soft/dtest-app.exe $(TOP)/mcs/class/Mono.Debugger.Soft/dtest-app.pdb
-$(TOP)/mcs/class/Mono.Debugger.Soft/dtest-app.exe $(TOP)/mcs/class/Mono.Debugger.Soft/dtest-app.pdb:
-	$(MAKE) PROFILE='monodroid' -C $(TOP)/mcs/class/Mono.Debugger.Soft dtest-app.exe
+.PHONY: $(TOP)/mcs/class/lib/monodroid/tests/dtest-app.exe $(TOP)/mcs/class/lib/monodroid/tests/dtest-app.pdb
+$(TOP)/mcs/class/lib/monodroid/tests/dtest-app.exe $(TOP)/mcs/class/lib/monodroid/tests/dtest-app.pdb:
+	$(MAKE) PROFILE='monodroid_tools' -C $(TOP)/mcs/class/Mono.Debugger.Soft test
```

Comment:
This is broken: you are building the `monodroid_tools` profile but picking the `monodroid` one.

Reply:
That's what you told me you want to happen. dtest-app.exe should be built against the monodroid profile since it'll run on the device.

I added the necessary magic to the Mono.Debugger.Soft Makefile so that this will do the right thing.

Reply:
There is confusion here: we need `dtest-app.exe` to come from `monodroid`, and `Mono.Debugger.Soft` to come from `monodroid_tools`, nothing else, nothing more. Here you are "building" in `monodroid_tools` but copying from `monodroid` which goes against any proper assumption. Also, it doesn't seem warranted from what I understand from https://github.com/mono/mono/pull/11033/files#diff-ea8be71a4e802c7bf5e99a96fc5a2ff0R11 since `dtest-app.exe` is a very simple "console" app which doesn't rely on anything else than `mscorlib` and `System` - which *are* available in `monodroid`. `Mono.Debugger.Soft` is, as we indeed discussed, a different beast since it depends on `Mono.Cecil` and others, which warrants that we build it in `monodroid_tools`.